### PR TITLE
Enable filtering out values based on location scale values

### PR DESCRIPTION
**Pull request by ghop02**
Need to go through and clean up tests, but this allows us some flexibility to report values if at least 95% of the population is reporting

### CODE REVIEW

File: libs/datasets/timeseries.py [hunk 1]
```diff
@@ -1098,29 +1119,19 @@ def _aggregate_dataframe_by_region(
...
-    if not ignore_na:
-        # Only keep aggregated values where the count of aggregated values is the same as the
-        # count of input regions.
-        long_agg = long_agg.loc[long_agg["count"] == long_agg["location_id_agg_count"]]
+    if reporting_ratio_required:
+        weighted_reporting_ratio = _calculate_weighted_reporting_ratio(
+            long_all_values, location_id_map, reporting_ratio_location_weights, groupby_columns
+        )
+        is_valid_reporting_ratio = weighted_reporting_ratio >= reporting_ratio_required
+        acceptable_ratios = weighted_reporting_ratio.loc[is_valid_reporting_ratio]
```
Comment (TomGoBravo) [inline libs/datasets/timeseries.py:1130]: not sure I'm following all the index but not sure you need acceptable_ratios if is_valid_reporting_ratio is a mask of locations with good ratios. How about

long_agg = long_agg.loc[is_valid_reporting_ratio]

File: libs/datasets/timeseries.py [hunk 2]
```diff
@@ -1134,9 +1145,20 @@ def aggregate_regions(
...
 ) -> MultiRegionDataset:
-    """Produces a dataset with dataset_in aggregated using sum or weighted aggregation."""
+    """Produces a dataset with dataset_in aggregated using sum or weighted aggregation.
+
+    Args:
+        dataset_in: Input dataset.
+        aggregate_map: Region mapping input region to aggregate region.
+        aggregations: Sequence of aggregation overrides to apply aggregations other
+            than sum to fields.
+        reporting_ratio_required_to_aggregate: Ratio of locations per aggregate region required
```
Comment (TomGoBravo) [inline libs/datasets/timeseries.py:1157]: Say something about the ratio of scaling factor metric.

File: libs/datasets/timeseries.py [hunk 3]
```diff
@@ -1134,9 +1145,20 @@ def aggregate_regions(
     aggregate_map: Mapping[Region, Region],
     aggregations: Sequence[StaticWeightedAverageAggregation] = WEIGHTED_AGGREGATIONS,
     *,
-    ignore_na: bool = False,
+    reporting_ratio_required_to_aggregate: Optional[float] = None,
```
Comment (TomGoBravo) [inline libs/datasets/timeseries.py:1148]: Why optional instead of float with default of 0.0 or 1.0?

**Review by TomGoBravo — COMMENTED** [review]
I'd like some tests of the edge cases.

File: libs/datasets/timeseries.py [hunk 4]
```diff
@@ -1174,12 +1186,18 @@ def aggregate_regions(
     # ... all other static input values.
     static_in_other_fields = static_in.loc[:, ~scale_fields_mask]
 
+    populations = static_in.loc[:, CommonFields.POPULATION]
```
[the remark this answers is not in the document]
Comment (ghop02) [reply libs/datasets/timeseries.py]: I think that's a next step if we want, but largely don't think it'll make that big of a difference

File: libs/datasets/timeseries.py (hunk 3 above)
Comment (ghop02) [reply libs/datasets/timeseries.py:1160]: I wanted to use none to specify that we're disabling filtering by ratio reported but can change it back

**Review by TomGoBravo — APPROVED** [review]
